### PR TITLE
Run autoconf from top-level directory.

### PR DESCRIPTION
**Pull request by coryan**
This is part (hopefully all) of the fixes for #776.  The top-level
configure.ac configures googletest first and then googlemock.
With these changes it is possible to embed googletest into another
project that uses autoconf.  For an example (though it is WIP), see
the commits (and soon PR) referenced from google/protobuf#236.

For example:
https://github.com/coryan/protobuf/tree/fix-issue-236-gmock-as-submodule

### CODE REVIEW

**Comment by gennadiycivil** [comment]
@coryan , could you please provide before/after test evidence that demonstrates how the issue is fixed. Thank you again for this contribution

**Comment by coryan** [comment]
PTAL.  With the latest commit I introduce a Travis build that works in this branch.  As shown [here](https://travis-ci.org/coryan/googletest/builds/321059480) the equivalent commit in `master` breaks.

File: .travis.yml [hunk 1]
```diff
@@ -24,10 +24,11 @@ matrix:
       script: ./ci/build-linux-bazel.sh
     - os: linux
       compiler: gcc
-      env: BUILD_TYPE=Debug VERBOSE=1
+      install: ./ci/install-linux.sh && ./ci/log-config.sh
+      script: ./ci/build-linux-autotools.sh
     - os: linux
       compiler: gcc
-      env: BUILD_TYPE=Release VERBOSE=1 CXX_FLAGS=-std=c++11
+      env: BUILD_TYPE=Debug VERBOSE=1 CXX_FLAGS=-std=c++11
```
Comment (gennadiycivil) [inline .travis.yml]: @coryan not sure why we are changing Release to Debug and how it is related to the running of automake from the top level dir?

File: googletest/Makefile.am [hunk 2]
```diff
@@ -120,16 +121,16 @@ EXTRA_DIST += \
...
-  msvc/gtest-md.vcproj \
-  msvc/gtest.sln \
-  msvc/gtest.vcproj \
-  msvc/gtest_main-md.vcproj \
-  msvc/gtest_main.vcproj \
-  msvc/gtest_prod_test-md.vcproj \
-  msvc/gtest_prod_test.vcproj \
-  msvc/gtest_unittest-md.vcproj \
-  msvc/gtest_unittest.vcproj
+  msvc/2010/gtest-md.sln \
```
Comment (gennadiycivil) [inline googletest/Makefile.am]: I would avoid adding vusual studion 2010 (!) ... why?

File: .travis.yml (hunk 1 above)
Comment (coryan) [reply .travis.yml]: That is just to keep the number of builds under control.  The autotools build already builds optimized Linux (Ubuntu) with g++, why have another build do the exact same thing?

File: googletest/Makefile.am (hunk 2 above)
Comment (coryan) [reply googletest/Makefile.am]: The files got moved from `msvc/*` to `msvc/2010/*` at some point, but `Makefile.am` was not changed at the same time.  If you think it would be best to remove the files altogether, then sure, but shouldn't that be a separate change?

File: .travis.yml (hunk 1 above)
Comment (coryan) [reply .travis.yml]: Happy to revert this change if you prefer.

Comment (gennadiycivil) [reply .travis.yml]: No problem, we can leave them this way, thank you

File: googletest/Makefile.am (hunk 2 above)
Comment (gennadiycivil) [reply googletest/Makefile.am]: Since we are making a change, I just want to understand why. Could you please take a look and suggest why this happened and more importantly is this the general enough way to make this change for all googletest users?

Comment (gennadiycivil) [reply googletest/Makefile.am]: @coryan

Comment (coryan) [reply googletest/Makefile.am]: > Since we are making a change, I just want to understand why.
> Could you please take a look and suggest why this happened

**TL;DR;** since there was no test for automake builds, nothing detected the breakage.

Let's play code archeologist:

The last time those lines in `Makefile.am` were touched was around 9 years ago:

https://github.com/google/googletest/blame/master/googletest/Makefile.am

At that time the `msvc/*` directory was full of files:

https://github.com/google/googletest/tree/bcaf6f542fac67549528027c24e530e0e89f700b/msvc

The files were removed in this commit:

https://github.com/google/googletest/commit/45287f3dffa494d612c4ba94199aeaa40c99571f

Which I think you merged here:

https://github.com/google/googletest/commit/f25921565bbfb16b02919165669734a5bea7ee98


> and more importantly is this the general enough way to make this change for all
> googletest users?

Well, those particular lines in `Makefile.am` are simply there to include the MSVC files in any tarballs created for a "distribution".  Whether you want to continue to support tarballs or not is a different question.  Right now automake insists on creating the tarball and testing the tarball.  I think if you want to drop support for tarballs that should be a separate change.  Or you may want to drop the MSVC files altogether, that is an option too, but those should be separate changes from "getting automake to work again".

Comment (gennadiycivil) [reply googletest/Makefile.am]: makes sense!
